azure-pipelines-build-stage-job.yml

```diff
@@ -72,16 +72,25 @@ jobs:
           MANUAL_MM_VERSION: true
         condition: or( eq( variables['Build.Reason'], 'PullRequest' ), eq( variables['Build.SourceBranchName'], 'mm2.1' ), eq( variables['Build.SourceBranchName'], 'dev' ) )
       - bash: |
-          zip upload/mm2-$(COMMIT_HASH)-$(Agent.OS)-Release target/release/mm2 target/release/libmm2.a -j
+          containers=$(docker ps -q | wc -l)
+          echo $containers
+          if [ $containers -gt 0 ]; then
+            docker rm -f $(docker ps -q)
+          fi
+        displayName: 'Clean up Docker containers'
+        # Run unconditionally even if previous steps failed
+        condition: true
+      - bash: |
+          zip upload/mm2-$(COMMIT_HASH)-$(Agent.OS)-Release target/release/mm2 -j
         displayName: 'Prepare release build upload Linux'
         condition: and ( eq( variables['Agent.OS'], 'Linux' ), ne ( variables['Build.Reason'], 'PullRequest' ) )
       - bash: |
           cd target/x86_64-apple-darwin/release
-          zip ../../../upload/mm2-$(COMMIT_HASH)-$(Agent.OS)-Release mm2.dSYM mm2 libmm2.a -r
```

Comment:
What's the reason for not uploading libmm2?

Reply:
libmm2 is big, takes a long time to zip, and seems to be not really used by anyone.